sktree/tree/manifold/_morf_splitter.pyx

```diff
@@ -281,7 +283,17 @@ cdef class BestPatchSplitter(BaseDensePatchSplitter):
         # (\prod_i self.patch_dims_buff[i])
         cdef SIZE_t patch_size
 
-        for proj_i in range(0, max_features):
+
+        ##### suposse randomly select one feature set to find a best split
+        cdef np.ndarray[int, ndim=1] choices = np.array([0, n_features_1])
```

Comment:
This doesn't work inside `nogil`. To instantiate an array requires the GIL. 

I thought what you are trying to do is pass in `n_features_1 = the # of features in dataset 1`. Then `X[:, :n_features_1]` is dataset 1 and `X[:, n_features_1:]` is dataset 2?

Reply:
yes! Sorry, I didn't check the git timely